ext/semian/resource.c

```diff
@@ -0,0 +1,262 @@
+#include "resource.h"
+
+// "Private" function forward declarations
```

Comment:
You don't really need to declare these IMO. Just define them before the the public functions. The fact that they're `static` conveys that they are "private".